packages/components/src/components/button/button.tsx

```diff
@@ -111,6 +111,9 @@ export class Button {
   connectedCallback() {
     this.setIconPositionProp();
     this.appendEnterKeySubmitFallback();
+    if (this.disabled) {
+      this.hostElement.removeAttribute('href');
+    }
```

Comment:
This can be declaratively, and also removing an attribute set by the user might be a bit rude haha, there's a suggestion below.